From 50b231ceb5d8d353970526a198f8b53ef6d249b9 Mon Sep 17 00:00:00 2001
From: Iaroslav Voitovych <yaroslav.voytovych@gmail.com>
Date: Sun, 18 Oct 2026 02:10:53 +0200
Subject: [PATCH BlueZ] emulator: Avoid variable length array in send_cmd

Use newa() for the iovec array of send_cmd(), as shared/gatt-client
does, instead of a variable length array.

This fixes the following warning:

emulator/btdev.c:479:29: warning: Variable length array is used.
---
 emulator/btdev.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/emulator/btdev.c b/emulator/btdev.c
index 67eac8fd0..7a8d8a464 100644
--- a/emulator/btdev.c
+++ b/emulator/btdev.c
@@ -476,7 +476,7 @@ static void send_cmd(struct btdev *btdev, uint8_t evt, uint16_t opcode,
 					const struct iovec *iov, int iovlen)
 {
 	struct bt_hci_evt_hdr hdr;
-	struct iovec iov2[2 + iovlen];
+	struct iovec *iov2 = newa(struct iovec, 2 + iovlen);
 	uint8_t pkt = BT_H4_EVT_PKT;
 	int i;
 

base-commit: fa6ba55020dee3c3183041c854c347b4582448be
-- 
2.43.0

